lesson14/main.js

```diff
@@ -0,0 +1,121 @@
+const overlay = document.getElementById("js-overlay");
+const openListButton = document.getElementById("js-open-list-button")
+const ul = document.getElementById("js-ul");
+const numberInput =document.getElementById("js-number-input")
+const validationMessage = document.getElementById("js-validation");
+
+const renderLoading = () => {
+  const loading = document.getElementById("js-loading");
+  const loadingImage = document.createElement("img");
+  loadingImage.src = "img/loading.gif";
+  loadingImage.id = "loading-image";
+  loading.appendChild(loadingImage);
+};
+
+const removeLoading = () => document.getElementById("loading-image").remove();
+
+const renderLists = (items) => {
+  const fragment = new DocumentFragment();
+
+  for (let item of items) {
+    const li = document.createElement("li");
+    const a = document.createElement("a");
+    const img = document.createElement("img");
+    a.href = item.to;
+    a.textContent = item.text;
+    img.src = item.img;
+    img.alt = item.alt;
+
+    fragment
+      .appendChild(li)
+      .appendChild(a)
+      .insertAdjacentElement("afterbegin", img);
+  }
+
+  ul.appendChild(fragment);
+};
+
+const displayInfo = (info) => {
+  const li = document.createElement('li');
+  if (typeof info === 'string') {
+    li.textContent = info;
+    ul.appendChild(li);
+    return console.log(info);
+  }
+  throw new Error('Please pass a string value');
+};
+
+const displayErrorInfo = (error) => {
+  const li = document.createElement('li');
+  li.textContent = `error: ${JSON.stringify(error)}`;
+  ul.appendChild(li);
+};
+
+const getData = async (api) => {
+  let res = null;
+  try {
+    res = await fetch(api);
+  } catch (e) {
+    displayErrorInfo(e);
+  }
+  if (res.ok) {
+    return res.json();
+  } else {
+    displayErrorInfo(res);
+  }
+};
+
+const displayList = async (userInput) => {
+  renderLoading();
+  let res = await getData("https://mocki.io/v1/ee8a871e-2b46-4a91-b565-4d6f9216f300");
+  removeLoading();
+
+  if (res.data) {
+    renderLists(res.data);
+    console.log(userInput)
+  } else {
+    displayInfo("no data");
+  }
+};
+
+const displayModal= () => {
+  modal.style.display= "block";
+  overlay.style.display= "block";
+};
+
+const closeModal = () => {
+  resetInput()
+  modal.style.display="none";
+  overlay.style.display="none";
+};
+
+const removeModal = () =>  document.getElementById("js-modal-wrapper").remove();
+
+const resetInput = () => {
+  numberInput.value = '';
+  validationMessage.style.display ="none";
+};
+
+openListButton.addEventListener("click", () => {
+  const userInput = numberInput.value;
+  if(userInput === ''){
+   validationMessage.textContent = "Input feild cannot be empty";
+   validationMessage.style.display="block";
+  } else if (!userInput.match(/^\d+$/)){
+    validationMessage.textContent = "Input value must be a number";
+    validationMessage.style.display="block";
+  } else {
+    removeModal();
+    displayList(userInput);
+  }
+});
```

Comment:
IMO
I found it easier to understand if statements than if else if statements. However, this may be a matter of preference.

After reading the following reference, I avoided using else if.

> It is equivalent to generating a nest of further if and else clauses in the else statement.

[ref](https://qiita.com/taiju_suzuki/items/e2bf11fcf1645623235f)

```suggestion
openListButton.addEventListener("click", () => {
  const userInput = numberInput.value;
  if (userInput === "") {
    validationMessage.textContent = "Input feild cannot be empty";
    validationMessage.style.display = "block";
    return;
  }

  if (!userInput.match(/^\d+$/)) {
    validationMessage.textContent = "Input value must be a number";
    validationMessage.style.display = "block";
    return;
  }
  removeModal();
  displayList(userInput);
});
```

--- 
I also agree with Mai's statement about functionalization.

Reply:
Thank you for sharing the article and your comment. 
It was a good to learn that else if can make the code complex and potentially cause a bug. 
I updated validation check function, please see the commit.
2932cfa8aa449e004cd37e005a3354402116c1df